packages/core/src/libraries/organization-invitation.ts

```diff
@@ -66,6 +66,25 @@ export class OrganizationInvitationLibrary {
 
     return this.queries.pool.transaction(async (connection) => {
       const organizationQueries = new OrganizationQueries(connection);
+      // Fetch existing invitations to the invitee first
+      const existingInvitations = await organizationQueries.invitations.findEntities({ invitee });
+      // Check if any pending invitation has expired, if yes, update the invitation status to "Expired" first
+      // Note: Even if the status may appear to be "Expired", the actual data in DB may still be "Pending".
+      // Check `findEntities` in `OrganizationQueries` for more details.
+      const expiringInvitations = existingInvitations.filter(
+        (invitation) => invitation.expiresAt < Date.now()
+      );
+      if (expiringInvitations.length > 0) {
+        await Promise.all(
+          expiringInvitations.map(async (invitation) =>
+            organizationQueries.invitations.updateById(invitation.id, {
+              status: OrganizationInvitationStatus.Expired,
+              updatedAt: Date.now(),
+            })
+          )
+        );
+      }
```

Comment:
you can directly update the invitation though.

Reply:
Updated with a new query method: `updateExpiredEntities`